cmd/updateIntegrationArtifactConfiguration_test.go

```diff
@@ -0,0 +1,90 @@
+package cmd
+
+import (
+	"testing"
+
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+type updateIntegrationArtifactConfigurationMockUtils struct {
+	*mock.ExecMockRunner
+	*mock.FilesMock
+}
+
+func newUpdateIntegrationArtifactConfigurationTestsUtils() updateIntegrationArtifactConfigurationMockUtils {
+	utils := updateIntegrationArtifactConfigurationMockUtils{
+		ExecMockRunner: &mock.ExecMockRunner{},
+		FilesMock:      &mock.FilesMock{},
+	}
+	return utils
+}
+
+func TestRunUpdateIntegrationArtifactConfiguration(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Successfully update of Integration Flow configuration parameter test", func(t *testing.T) {
+		config := updateIntegrationArtifactConfigurationOptions{
+			Host:                   "https://demo",
+			OAuthTokenProviderURL:  "https://demo/oauth/token",
+			Username:               "demouser",
+			Password:               "******",
+			IntegrationFlowID:      "flow1",
+			IntegrationFlowVersion: "1.0.1",
+			ParameterKey:           "myheader",
+			ParameterValue:         "def",
+		}
+
+		httpClient := httpMockCpis{CPIFunction: "UpdateIntegrationArtifactConfiguration", ResponseBody: ``, TestType: "Positive", Method: "PUT", URL: "https://demo/api/v1/IntegrationDesigntimeArtifacts(Id='flow1',Version='1.0.1')"}
+
+		err := runUpdateIntegrationArtifactConfiguration(&config, nil, &httpClient)
+
+		if assert.NoError(t, err) {
+
+			t.Run("check url", func(t *testing.T) {
+				assert.Equal(t, "https://demo/api/v1/IntegrationDesigntimeArtifacts(Id='flow1',Version='1.0.1')/$links/Configurations('myheader')", httpClient.URL)
+			})
+
+			t.Run("check method", func(t *testing.T) {
```

Comment:
What is the reason of encapsulating the assert in another test run?

Reply:
... and why not? With that we get a nice check-tree when calling `go test -v`. We can see from the test name what exactly failed.

Reply:
That was one thought I had, yes, that one gets another addition to the test description. For example in this case that the method is checked here.